Using --entryFile flag with webpack doesn't work

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When using `--entryFile` with `nest start` command to specify different file to run (`repl.ts` for example) doesn't work as intended when using `webpack` instead of `tsc`.

When using custom `webpack.config.js` in following manner to diagnose the issue:
```js
module.exports = options => {
  console.log(options.output.filename);
  console.log(options.entry);
  return options;
}
```

We get the following output:
```js
> nest start graphql --entryFile generate

apps/graphql/main.js
/workspace/apps/graphql/src/main.ts

webpack 5.79.0 compiled successfully in 1323 ms
Error: Cannot find module '/workspace/dist/apps/graphql/generate'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1021:15)
    at Function.Module._load (node:internal/modules/cjs/loader:866:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:22:47
```

Which indicates that `--entryFile` has no affect on webpack and it continues to compile the `main.ts` which is defined in `nest-cli.json` as `entryFile`.

Using `nest start --tsc --entryFile repl` works as intended.

### Minimum reproduction code

https://github.com/mentos1386/nest-cli-entryfile-repro

### Steps to reproduce

1. `npm i`
2. `npx nest start --entryFile repl`

### Expected behavior

`repl.ts` should be compiled and started.

### Package version

9.4.0

### NestJS version

9.4.0

### Node.js version

16

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Other

Relates to #1691

--- a/actions/build.action.ts
+++ b/actions/build.action.ts
@@ -124,6 +124,7 @@ export class BuildAction extends AbstractAction {
       );
       return this.webpackCompiler.run(
         configuration,
+        options,
         webpackConfigFactoryOrConfig,
         pathToTsconfig,
         appName,

--- a/lib/compiler/webpack-compiler.ts
+++ b/lib/compiler/webpack-compiler.ts
@@ -6,6 +6,7 @@ import { AssetsManager } from './assets-manager';
 import { webpackDefaultsFactory } from './defaults/webpack-defaults';
 import { getValueOrDefault } from './helpers/get-value-or-default';
 import { PluginsLoader } from './plugins-loader';
+import { Input } from '../../commands';
 import webpack = require('webpack');
 
 type WebpackConfigFactory = (
@@ -22,6 +23,7 @@ export class WebpackCompiler {
 
   public run(
     configuration: Required<Configuration>,
+    options: Input[],
     webpackConfigFactoryOrConfig:
       | WebpackConfigFactoryOrConfig
       | WebpackConfigFactoryOrConfig[],
@@ -51,6 +53,8 @@ export class WebpackCompiler {
       configuration,
       'sourceRoot',
       appName,
+      'sourceRoot',
+      options,
     );
     const pathToSource =
       normalize(sourceRoot).indexOf(normalize(relativeRootPath)) >= 0
@@ -61,6 +65,8 @@ export class WebpackCompiler {
       configuration,
       'entryFile',
       appName,
+      'entryFile',
+      options,
     );
     const entryFileRoot =
       getValueOrDefault<string>(configuration, 'root', appName) || '';

